### ISSUE TITLE
Changing question doesn't scroll new question into viewport

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#2848]
## Bug Description

On the User Input screen, when the Back/Next buttons are clicked the viewport doesn't change. This is a problem on smaller screens (eg mobile) where advancing to the next question causes it to appear "above the fold". We should scroll the viewport to the new question whenever we change the "active" question.

It's worth noting this happens—at least in part—because we don't use a dedicated router (see #2514). I think #2514 would fix this issue as React Router is good about scrolling new content from a route change when it isn't visible. And there are more predictable ways to get the content to scroll into view with a router.

So it might be best to wait on this until #2514 is fixed, to see if it covers the issue.

## Steps to reproduce

1. Go to the User Input questionnaire on a new site where the answers have not been submitted.
2. Using a small/mobile viewport, scroll to the bottom of the first question, select an answer, then click "Next".
3. Notice the question is outside the viewport and the screen needs to be scrolled up to read it.

## Screenshots

![2021-02-22 22 48 33](https://user-images.githubusercontent.com/90871/108780429-54814400-7560-11eb-98d7-047bb0dfd1e5.gif)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The next question should clearly appear after the "Next" button is clicked in the user input screen.
* The viewport should be scrolled so the _beginning_ of the new question component is in view, not the end of the component.

## Implementation Brief

* Add <code>id={ \`googlesitekit-user-input-question-${ questionNumber }\` }</code> to the top-most `div` in `assets/js/components/user-input/UserInputQuestionWrapper.js`
* Add a `useEffect()` hook in `UserInputQuestionnaire` that scrolls to the above `googlesitekit-user-input-question-${ questionNumber }` (eg `googlesitekit-user-input-question-${ activeSlugIndex + 1 }`) element after the first render when `activeSlugIndex` changes.
* Merge #2912

### Test Coverage

* No change

### Visual Regression Changes

* N/A

## QA Brief

* Follow the steps to reproduce above, but note that the question is fully visible after being scrolled into view when clicking "Next" or "Back".

## Changelog entry

* Fix bug where switching between questions in the user input flow did not scroll the question into view.

**Comment by aaemnnosttv** [comment]
@tofumatt #2514 doesn't touch anything for user input. I saw you mentioned it in the description of that issue, but the ACs are limited to `SettingsApp` only. We should probably implement a temporary solution here (which shouldn't be too much effort I would think) and open a new issue to refactor user input with the router once it's merged.

**Comment by aaemnnosttv** [comment]
@tofumatt I think the added `id` is unnecessary if we can get a `ref` to the element.

I think we could do it entirely with a function ref since it will be called when it gets the element (or unmounts)

```js
<UserInputQuestionWrapper
    ref={ isActive ? ( el ) => el?.scrollIntoView( true ) : undefined }
/>
```

What do you think?

**Comment by tofumatt** [comment]
The issue with that approach is it scrolls the element into view on page load:

![2021-03-12 18 48 31](https://user-images.githubusercontent.com/90871/110984965-b8c74480-8363-11eb-8413-09d6d99b58a5.gif)

So anything above the question element is immediately scrolled out of view.

**Comment by aaemnnosttv** [comment]
IB ✅ 

Please don't forget to fill in the test and VRT sections of the IB though. In this case I think they're not needed but we should always be explicit about that.

**Comment by wpdarren** [comment]
### QA Update: Fail ❌ 

@tofumatt  The initial issue is fixed but thought I would highlight these two observations: when you edit a question, make a change and then click the `update` button, you are taken back to the preview, but it loads towards the bottom of the page.  

This is possibly linked with the issue above, but a different scenario:

I noticed when you get to the last question (when completing it for the first time) and you click on the preview button.  Rather than jumping to the top of the page, it remains towards the bottom of the page. Like above. 

You then have to scroll to the top to preview all the answers. 

![image](https://user-images.githubusercontent.com/73545194/111778608-e2e2ae80-88ef-11eb-807d-47ae0d26c841.png)

**Comment by tofumatt** [comment]
Ah, I guess that's a similar issue but not exactly unrelated. My instinct is I'd want the completed form to be near submit, but that doesn't happen here—it just stays in the previous position.

I guess we should scroll up to the top of the page when the summary is loaded, so I'll work on that. Thanks 👍🏻

**Comment by eugene-manuilov** [comment]
@aaemnnosttv or @tofumatt do you mind reviewing #3008? It fixes styles issues and one bug in the search term input fields.

**Comment by aaemnnosttv** [comment]
@eugene-manuilov does your PR address everything in https://github.com/google/site-kit-wp/issues/2848#issuecomment-802794834? Otherwise, looks good.

**Comment by eugene-manuilov** [comment]
@aaemnnosttv #2995 fixed everything in QA feedback comment. #3008 fixes a few more issues found during testing #2995.

**Comment by wpdarren** [comment]
### QA Update: Fail ❌ 

@eugene-manuilov @tofumatt there are still a few issues, I am afraid. 

1) The two issues reported in my [previous comment](https://github.com/google/site-kit-wp/issues/2848#issuecomment-802794834) do not appear to be fixed. Refreshed develop branch. Only applies to small sized viewpoints. 

2) The initial issue reported with the scrolling is resolved on small viewpoints but unfortunately, it's now an issue on desktop. When clicking "Next" or "Back" buttons,  the scrolling does not go to the top and question is not fully visible. 

**Verified:**

On small sized viewpoints the question is fully visible after being scrolled into view when clicking "Next" or "Back" buttons.

**Comment by wpdarren** [comment]
### QA Update: Pass ✅ 

**Verified:**

- The question is fully visible after being scrolled into view when clicking "Next" or "Back" on desktop and mobile. 
- Made sure that when preview was clicked the page scrolled automatically goes to the top off the questions
- Made sure that when the edit link is clicked and a question changed then updated, the page scrolled automatically goes to the top off the questions

Looking great, nice work @tofumatt!
